fhir2dataset/api_caller.py

```diff
@@ -178,14 +177,17 @@ def _get_data(self):
         """retrieves the necessary information from the json instance of a resource and stores it in the data attribute
         """  # noqa
         elements_empty = asdict(self.elements)
-        for json_resource in self.results:
-
-            data_dict = multiple_search_dict([json_resource["resource"]], elements_empty)[
-                0
-            ]  # because there is only one resource
-            elements = from_dict(data_class=Elements, data=data_dict)
-            df = self._flatten_item_results(elements)
-            self.df = pd.concat([self.df, df])
+        if not self.results:
+            columns = [element.col_name for element in self.elements.elements]
+            self.df = pd.DataFrame(columns=columns)
+        else:
+            for json_resource in self.results:
+                data_dict = multiple_search_dict([json_resource["resource"]], elements_empty)[
+                    0
+                ]  # because there is only one resource
```

Comment:
Could we use multiple_search_dict on all the resources simultaneously?

Reply:
Yes thank you, it was implemented, what do you think of it ?